<a href="https://colab.research.google.com/github/J-Salt/IDS/blob/main/HW4/problem2_r.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Importing the data
---

In [ ]:
library(readr)
nutrition_subset <- read_csv("https://raw.githubusercontent.com/J-Salt/IDS/main/HW4/nutrition_subset.csv", show_col_types = FALSE)
head(nutrition_subset)

food item                              weight_in_grams saturated_fat
1 GELATIN; DRY                  1 ENVELP   7.00           0.0         
2 SEAWEED; SPIRULINA; DRIED     1 OZ      28.35           0.8         
3 YEAST; BAKERS; DRY; ACTIVE    1 PKG      7.00           0.0         
4 PARMESAN CHEESE; GRATED       1 OZ      28.35           5.4         
5 PARMESAN CHEESE; GRATED       1 CUP    100.00          19.1         
6 PARMESAN CHEESE; GRATED       1 TBSP     5.00           1.0         
  cholesterol
1  0         
2  0         
3  0         
4 22         
5 79         
6  4

#Printing the 5 foods highest in saturated fat
---

In [ ]:
head(nutrition_subset[order(nutrition_subset$saturated_fat, decreasing = TRUE),], 5)

food item                             weight_in_grams saturated_fat
1 CHEESECAKE                    1 CAKE  1110            119.9        
2 ICE CREAM; VANLLA; RICH 16% FT1/2 GAL 1188            118.3        
3 YELLOWCAKE W/ CHOCFRSTNG;COMML1 CAKE  1108             92.0        
4 CREME PIE                     1 PIE    910             90.1        
5 LARD                          1 CUP    205             80.4        
  cholesterol sat_fat_per_gram
1 2053        0.10801802      
2  703        0.09957912      
3  609        0.08303249      
4   46        0.09901099      
5  195        0.39219512

#Printing the 5 foods with the highest saturated fat per gram
---

In [ ]:
nutrition_subset$sat_fat_per_gram <- nutrition_subset$saturated_fat / nutrition_subset$weight_in_grams
head(nutrition_subset[order(nutrition_subset$sat_fat_per_gram, decreasing = TRUE),],5)

food item                             weight_in_grams saturated_fat
1 BUTTER; SALTED                1 TBSP   14              7.1         
2 BUTTER; UNSALTED              1 TBSP   14              7.1         
3 BUTTER; SALTED                1/2 CUP 113             57.1         
4 BUTTER; UNSALTED              1/2 CUP 113             57.1         
5 BUTTER; SALTED                1 PAT     5              2.5         
  cholesterol sat_fat_per_gram
1  31         0.5071429       
2  31         0.5071429       
3 247         0.5053097       
4 247         0.5053097       
5  11         0.5000000

#Using ZScore to find outliers in saturated fat per gram
---

In [ ]:
nutrition_subset$scaled_sat_fat_per_gram <- scale(x = nutrition_subset$sat_fat_per_gram)

three_sigma_rule <- (nutrition_subset$scaled_sat_fat_per_gram < -3 | 
                     nutrition_subset$scaled_sat_fat_per_gram > 3 )



cat ("There are", sum(three_sigma_rule), "outliers \n\n")



cat("The scaled values that are outliers: \n\n")
nutrition_subset$scaled_sat_fat_per_gram[which(three_sigma_rule)]

summary(nutrition_subset$scaled_sat_fat_per_gram)
summary(nutrition_subset$sat_fat_per_gram)




There are 15 outliers 

The scaled values that are outliers: 



[1] 4.238469 3.936637 4.202078 3.940837 7.079055 7.079055 5.368580 3.276520
 [9] 5.370282 3.222049 7.106775 7.106775 6.998763 6.998763 4.730522

       V1         
 Min.   :-0.5620  
 1st Qu.:-0.5620  
 Median :-0.4411  
 Mean   : 0.0000  
 3rd Qu.: 0.1539  
 Max.   : 7.1068  

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.00000 0.00800 0.03717 0.04735 0.50714 

#Using ZScore to scale the other numerical values
---

In [ ]:
#Zero mean the numerical values
#Cholesterol
nutrition_subset$scaled_cholesterol <- scale(x = nutrition_subset$cholesterol)

three_sigma_rule <- (nutrition_subset$scaled_cholesterol < -3 | 
                     nutrition_subset$scaled_cholesterol > 3 )



cat ("There are", sum(three_sigma_rule), "outliers \n\n")



cat("The scaled values that are outliers: \n\n")
nutrition_subset$scaled_cholesterol[which(three_sigma_rule)]

summary(nutrition_subset$scaled_cholesterol)
summary(nutrition_subset$cholesterol)

#Weight
nutrition_subset$scaled_weight_in_grams <- scale(x = nutrition_subset$weight_in_grams)

three_sigma_rule <- (nutrition_subset$scaled_weight_in_grams < -3 | 
                     nutrition_subset$scaled_weight_in_grams > 3 )



cat ("There are", sum(three_sigma_rule), "outliers \n\n")



cat("The scaled values that are outliers: \n\n")
nutrition_subset$scaled_weight_in_grams[which(three_sigma_rule)]

summary(nutrition_subset$scaled_weight_in_grams)
summary(nutrition_subset$weight_in_grams)


There are 17 outliers 

The scaled values that are outliers: 



[1]  3.146420  4.355150  8.148063  5.063716 16.842585  8.898309  4.471855
 [8]  4.330142  4.713601  9.590203  4.530208  5.188757  6.872644  4.805298
[15]  3.696601  5.030372  5.588889

       V1         
 Min.   :-0.2714  
 1st Qu.:-0.2714  
 Median :-0.2714  
 Mean   : 0.0000  
 3rd Qu.:-0.1046  
 Max.   :16.8426  

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00    0.00    0.00   32.55   20.00 2053.00 

There are 22 outliers 

The scaled values that are outliers: 



[1] 4.323765 6.887585 5.460714 3.840561 3.567693 5.443659 7.882415 5.449344
 [9] 4.323765 5.108259 5.631256 3.925832 5.449344 5.199215 5.381127 5.904124
[17] 4.522731 4.522731 4.522731 4.522731 4.323765 7.916524

       V1         
 Min.   :-0.8471  
 1st Qu.:-0.6674  
 Median :-0.2070  
 Mean   : 0.0000  
 3rd Qu.: 0.3899  
 Max.   : 7.9165  

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    0.4    32.0   113.0   149.4   218.0  1542.0 

#Summary
---
By comparing foods, especially on this list, by just saturated fat content isn't a valid measure of unhealthy foods. This is because this doesn't account for the size of the food which could be an entire cake. The food with the highest saturated fat per gram had 0.50714 grams of saturated fat per gram of food. There were 15 outliers in the saturated fat per gram. These all came from the high end of the scale, and none were outside the lower bounds of 3 standard deviations. I did not find any missing values as I could not interpret zeros in cholesterol or saturated fat as missing since its possible for foods to have zero of both. I did check to make sure everything had a weight and that was true so therefore I did not have to deal with missing values. Lastly, I found 17 cholesterol outliers and 22 weight outliers in this data set. They were all above 3 standard deviations from the mean.